java-checks/src/test/files/checks/AssertionFailInCatchBlockCheck.java

```diff
@@ -0,0 +1,84 @@
+import org.junit.Assert;
+import org.junit.Test;
+
+public class MyTest {
+
+  static {
+    try {
+      File.createTempFile(null, null);
+    } catch (IOException e) {
+      Assert.fail(e.getMessage()); // Compliant, corner case accepted ... and MethodMatcher doesn't pass through this case
```

Comment:
This whole test file lacks at least one precise issue location : see for instance https://github.com/SonarSource/sonar-java/blob/master/java-checks/src/test/files/checks/naming/BadMethodName.java#L5 with the [[sc=;ec=]] syntax 

This should be done for one message at least to prevent regression when rule will be improved eventually.